lib/pure/parseutils.nim

```diff
@@ -242,7 +244,7 @@ proc parseIdent*(s: string, start = 0): string =
     while i < s.len and s[i] in IdentChars: inc(i)
     result = substr(s, start, i-1)
 
-proc parseChar*(s: string, c: var char, start = 0): int =
+proc parseChar*(s: openArray[char], c: var char, start = 0): int =
```

Comment:
openArray allows for effective slicing making the `start` parameters redundant and weird.